File: Auxiliary/gclib/examples/vb/examples/examples/Vector_Mode_Example.vb
Module Vector_Mode_Example

    Function Main() As Integer
        Dim rc As Integer = Examples.GALIL_EXAMPLE_OK
        Dim gclib As Gclib = New Gclib()
        Dim args() As String = Environment.GetCommandLineArgs()
        Try
            If args.Count() <> 3 Then
                Console.WriteLine("Incorrect number of arguments provided")
                Console.WriteLine("Usage: vector_mode_example.exe <ADDRESS> <FILE>")

                Console.Write(vbCrLf + "Press any key to close the example.")
                Console.ReadKey()
                Return Examples.GALIL_EXAMPLE_ERROR
            End If

            Dim address As String = args(1) 'Retrieve address from command line
            Dim file = args(2) 'Retrieve file from command line

            gclib.GOpen(address)

            rc = Examples.Vector_Mode(gclib, file)
        Catch ex As Exception
            Examples.PrintError(gclib, ex)
            rc = Examples.GALIL_EXAMPLE_ERROR
        Finally
            gclib.GClose()
        End Try

        Console.Write(vbCrLf + "Press any key to close the example.")
        Console.ReadKey()

        Return rc
    End Function

End Module

File: Auxiliary/gclib/examples/cs/examples/examples/vector_mode_example.cs
/** @addtogroup cs_examples
  * @{
  */
  
/*! \file vector_mode_example.cs
*
* See \link examples.Examples.Vector_Mode() Vector_Mode() \endlink for implementation of logic
* <br><br>For VB.NET, see definition in file vector_mode_example.vb
*/
using System;
using System.Linq;

namespace examples
{
    /** @addtogroup cs_examples
    * @{
    */
	/// <summary>
	/// Takes a path to a file at the command line holding vector commands for the controller.
	/// The controller is placed into vector mode and commands are read from the file and sent to
	/// the controller.
	/// </summary>
	/// <remarks>The first argument should be the IP Address of a Galil controller.
	/// The second argument should be a path to a file containing vector commands.</remarks>
	/*! For VB.NET, see definition in file vector_mode_example.vb */
    class Vector_Mode_Example
    {
        /// <summary>
        /// Main function for the vector mode example.
        /// </summary>
        /// <param name="args">An array of command line arguments.</param>
        /// <returns>The success status or error code of the function.</returns>
        /// <remarks>The first argument should be the IP Address of a Galil controller.
        /// The second argument should be a path to a file containing vector commands.</remarks>
        public static int Main(string[] args)
        {
            gclib gclib = new gclib();
            int rc = Examples.GALIL_EXAMPLE_OK;

            try
            {
                if (args.Count() != 2)
                {
                    Console.WriteLine("Incorrect number of arguments provided");
                    Console.WriteLine("Usage: Vector_Example.exe <ADDRESS> <FILE>");

                    Console.Write("\nPress any key to close the example");
                    Console.ReadKey();
                    return Examples.GALIL_EXAMPLE_ERROR;
                }

                string file = args[1]; //Retrieve file from command line
                string address = args[0]; //Retrieve address from command line
                gclib.GOpen(address); //Open a connection at the provided address

                rc = Examples.Vector_Mode(gclib, file);
            }
            catch(Exception ex)
            {
                Examples.PrintError(gclib, ex);
                rc = Examples.GALIL_EXAMPLE_ERROR;
            }
            finally
            {
                gclib.GClose();
            }

            Console.Write("\nPress any key to close the example");
            Console.ReadKey();
            return rc;
        }
    }
/** @}*/
}
/** @}*/